src/asn1/asn1-cer.cc

```diff
@@ -72,76 +73,94 @@ std::string cer_encode_string(const std::string& contents, ASN1_UniversalTag tag
 }
 
 static int32_t cer_decode_length(const std::string& content, int32_t* endpos) {
-  if (((uint8_t)content[0]) == 0x80) {
+  if ((static_cast<uint8_t>(content[0])) == 0x80) {
     *endpos = 1;
     return -1;
   }
 
-  if (!(((uint8_t)content[0]) & 0x80)) {
+  if (((static_cast<uint8_t>(content[0])) & 0x80) == 0) {
     *endpos = 1;
     return content[0];
   }
 
-  int len_len = ((uint8_t)content[0]) & 0x7F;
+  int len_len = (static_cast<uint8_t>(content[0])) & 0x7F;
   *endpos = len_len + 1;
-  if (len_len > 4) return -2;
+  if (len_len > 4) {
+    return -2;
+  }
 
   int32_t res = 0;
-  for (int i = 0; i < len_len; i++) {
+  for (uint64_t i = 0; i < len_len; i++) {
```

Comment:
If `len_len` is an `int`, then so should `i`. Isn't it best to compare two values of the same type?

Reply:
Yes, You are right

Reply:
changed len_len to uin64_t